src/sql-parser/src/ast/defs/statement.rs

```diff
@@ -323,20 +323,22 @@ impl AstDisplay for CopyDirection {
 impl_display!(CopyDirection);
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum CopyTarget {
+pub enum CopyTarget<T: AstInfo> {
     Stdin,
     Stdout,
+    Url(Expr<T>),
```

Comment:
In the case of S3, it's not the absolute name of the file, but rather should be a path to an empty directory (depending upon the parallelism and the max file size specified, we would end up writing multiple files). The `filename` parameter implies otherwise though?

Reply:
For postgres, COPY TO|FROM can use local file paths. But that's never going to be the case for Materialize and it will always have to be a url, which can either point to a file or a directory.

Also, from the postgres syntax link,
> There is no COPY statement in the SQL standard.

Would it make sense to actually keep `url` instead of `filename`?

Reply:
I agree that we'll never support the case of local files. My argument about having the parser/ast call this filename is that's the term postgres uses for this syntax. Planning/sequencing should be the things that take the filename syntax and start calling it a url or whatever. The reasoning is: because we are using an existing postgres syntax, the parser should match it exactly. If we were inventing a new syntax we could call it URL, but we aren't.

Reply:
There’s a third option here, which is to call the variant `Expr(Expr)`. That way the parser makes no particular implication as to the meaning of the expression.

Reply:
Concur about `Expr`! I realized last night that because we are supporting expressions, we are already deviating from postgres which supports only a string literal in this position.

Reply:
Great, updating this to `Expr` 👍

Reply:
Done!